### ISSUE TITLE
Data Table Expand Row needs the ability to add text (hidden or visual) to table header

### ISSUE THREAD

**Issue by sandraclarkdev** [carbon-design-system/carbon#5028]
## Environment
> Operating system
Windows

> Browser\
Chrome/FireFox/IE

> Automated testing tool and ruleset

> Assistive technology used to verify
WAVE/Jaws

## Detailed description
When using <TableExpandedRow><TableExpandedHeader> there should be a way to add either a visual or hidden text for screen readers "Expanded" to allow that row to have a non-empty header.

Violates WCAG 2.0/2.1 1.3.1 Info and Relationships (Level A)

> What version of the Carbon Design System are you using?
10.3.0 / React 7.3.0

> What did you expect to happen?
That the <TableExpandHeader would give me the ability to add props for "setHeaderText("string") and showHeaderText(true/false). If setHeaderText is not empty, it would either visually show the text on the column header or if false, arrange that it is positioned off-left and readable by screen readers.

> What happened instead?

> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

1. Set up a Data Table with a TableExpandedRow 
2. Verify page in WAVE
3. Note that WAVE fails for Empty table header
4. etc.

> Please create a reduced test case in CodeSandbox
https://2bvnv.csb.app/

## Additional information

- Screenshots or code
![image](https://user-images.githubusercontent.com/58484559/72353269-c296de80-36b1-11ea-8b4e-5ce7b621e0ae.png)

- Notes
